delira/utils/config.py

```diff
@@ -513,6 +515,135 @@ def create_from_str(cls, data, formatter=yaml.load, decoder_cls=Decoder,
+            for key, val in dict_like.items():
+                name = "--{}".format(prefix + key)
+                if val is None:
+                    parser.add_argument(name)
+                else:
+                    if type(val) == bool:
+                        parser.add_argument(name, type=str2bool)
+                    elif isinstance(val, (list, tuple)):
+                        if len(val) > 0 and type(val[0]) != type:
+                            parser.add_argument(name, type=type(val[0]))
+                        else:
+                            parser.add_argument(name)
+                    elif isinstance(val, dict):
+                        add_val(val, prefix=key + '.')
+                    elif isinstance(val, type):
+                        parser.add_argument(name, type=val)
+                    else:
+                        parser.add_argument(name, type=type(val))
+
+        add_val(self)
+        return parser
+
+    @staticmethod
+    def _add_unknown_args(unknown_args):
+        '''
+        Can add unknown args as parsed by argparsers method
+        `parse_unknown_args`.
+
+        Parameters
+        ------
+        unknown_args : list
+            list of unknown args
+        Returns
+        ------
+        Config
+            a config of the parsed args
+        '''
+        # first element in the list must be a key
+        assert unknown_args[0].startswith('--')
+
+        args = Config()
+        # take first key
+        key = unknown_args[0][2:]
+        idx, done, val = 1, False, []
+        while not done:
+            try:
+                item = unknown_args[idx]
+            except IndexError:
+                done = True
+            if item.startswith('--') or done:
+                # save key with its value
+                if len(val) == 0:
+                    # key is used as flag
+                    args[key] = True
+                elif len(val) == 1:
+                    args[key] = val[0]
+                else:
+                    args[key] = val
+                # new key and flush data
+                key = item[2:]
+                val = []
+            else:
+                val.append(item)
+            idx += 1
+        return args
+
+    def update_from_argparse(self, parser=None, add_unknown_items=False):
+        '''
+        Updates the config with all values from the command line.
+        Following the pattern: `--training.learning_rate 1234`
+
+        Raises
+        ------
+        TypeError
+            raised if another datatype than currently in the config is parsed
+        Returns
+        -------
+        dict
+            dictionary containing only updated arguments
+        '''
+
+        if len(sys.argv) > 1:
+            if not parser:
+                parser = self.create_argparser()
+
+            params, unknown = parser.parse_known_args()
+            params = vars(params)
+            if len(unknown) > 0 and add_unknown_items:
```

Comment:
I think this condition must be changed to `if unknown and not add_unknown_items` to reflect the actual state reflected in the warning below. (Note that there is now need to include `>0` or even the length method in this condition since `bool(len(iterable)>0) == bool(len(iterable)) == bool(iterable)`

Reply:
You are right!